src/encoded/static/components/experiment_series.js

```diff
@@ -543,26 +555,19 @@ class ExperimentSeriesComponent extends React.Component {
 
             // Experiment table data
             Object.keys(viewableDatasets).forEach((datasetAtId) => {
-                // Select one analysis based on file assembly and file count
-                // for extracting quality metrics
-                const filesByDesiredAssembly = [];
-                viewableDatasets[datasetAtId].files.forEach((f) => {
-                    if (selectedAssembly.includes(f.assembly)) {
-                        filesByDesiredAssembly.push(f['@id']);
-                    }
-                });
+                // Select one analysis based on analysis award
                 const analysisObjects = viewableDatasets[datasetAtId].analysis_objects || [];
-                const selectedAnalysis = analysisObjects.filter(
+                viewableDatasets[datasetAtId].selectedAnalysis = analysisObjects.filter(
```

Comment:
We’ll have to be careful going forward, as ESLint/AirBnB get more strict about mutating objects that don’t belong to us. I’m not sure if the datasets in viewableDatasets comprise copies of incoming datasets, or comprise the original datasets from React. If the latter, we shouldn’t mutate them. But in this case, I don’t remember anymore.

Reply:
I'm not sure if I understand `belong to us` correctly. So `viewableDatasets` is a state from this method and is build from a series of requests: https://github.com/ENCODE-DCC/encoded/blob/dev/src/encoded/static/components/experiment_series.js#L417-L449 . So maybe it belongs to the component thus belong to us?

Reply:
If the contents of viewableDatasets includes references to the datasets coming through props.context, then those objects belong to React and we shouldn't mutate them. Problem is, we've committed this mistake often in the past, and I don't remember if viewableDatasets makes copies of the datasets objects (so we can mutate those) or if it references props.context and we shouldn't, and it wasn't obvious by looking which was the case.